app/services/api/to_user_sessions_hash2.rb

```diff
@@ -0,0 +1,73 @@
+class Api::ToUserSessionsHash2
```

Comment:
Maybe you could think of some refactoring here. Don't have any ideas at the moment. But available to sit together if you want

Reply:
You can always extract services if that helps. Prolly not, I feel like the problem here is that we are fighting against OOP :)